code/datums/components/seethrough_mob.dm

```diff
@@ -0,0 +1,124 @@
+///A component that lets you turn your character transparent in order to see and click through yourself.
+/datum/component/seethrough_mob
+	///The atom that enables our dark magic
+	var/obj/render_source_atom
+	///The fake version of ourselves
+	var/image/trickery_image
+	///Which alpha do we animate towards?
+	var/target_alpha
+	///How long our faze in/out takes
+	var/animation_time
+	///Does this object let clicks from players its transparent to pass through it
+	var/clickthrough
+	///Is the seethrough effect currently active
+	var/is_active
+
+/datum/component/seethrough_mob/Initialize(target_alpha = 100, animation_time = 0.5 SECONDS, clickthrough = TRUE)
+	. = ..()
+
+	if(!ismob(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.target_alpha = target_alpha
+	src.animation_time = animation_time
+	src.clickthrough = clickthrough
+	src.is_active = FALSE
+	var/datum/action/toggle_seethrough/action = new(src)
+	action.Grant(parent)
+
+///Set up everything we need to trick the client and keep it looking normal for everyone else
+/datum/component/seethrough_mob/proc/trick_mob()
+	SIGNAL_HANDLER
+
+	var/mob/fool = parent
+	var/datum/hud/our_hud = fool.hud_used
+	for(var/atom/movable/screen/plane_master/seethrough in our_hud.get_true_plane_masters(SEETHROUGH_PLANE))
+		seethrough.unhide_plane(fool)
+
+	render_source_atom = new()
```

Comment:
It would be less wasteful to instantiate and delete this once, instead of everytime.